engine/datastore/models.go

```diff
@@ -371,15 +370,12 @@ func (c *Client) Raw(query string) *gorm.DB {
 // checkResult will check for records or error
 func checkResult(result *gorm.DB) error {
 	if result.Error != nil {
-		if errors.Is(result.Error, gorm.ErrRecordNotFound) {
-			return ErrNoResults
-		}
 		return result.Error
 	}
 
 	// We should actually have some rows according to GORM
 	if result.RowsAffected == 0 {
-		return ErrNoResults
+		return gorm.ErrRecordNotFound
```

Comment:
We previously returned a custom error rather then gorms. Now that in this pr i've introduced translation of errors to gorm we need to return a gorm not found error. This will be checked in other places for that error